Client/multiplayer_sa/CMultiplayerSA_WidescreenFix.cpp

```diff
@@ -0,0 +1,1111 @@
+        0x5831AD,            // 21 Legend
+        0x57613B,            // 22 Legend
+        0x575EDF,            // 23 Map zones
+        0x57A88D,            // 24 Brightness slider posn
+        0x57AA81,            // 25 Radio slider posn
+        0x57ACBE,            // 26 Sfx slider posn
+        0x57AEB0,            // 27 Draw slider posn
+        0x57B0E4,            // 28 Mouse slider posn
+        0x57A807,            // 29 Brightness slider
+        0x57A811,            // 30 Brightness slider
+        0x57A9FE,            // 31 Radio slider
+        0x57AA08,            // 32 Radio slider
+        0x57AC2F,            // 33 Sfx slider
+        0x57AC39,            // 34 Sfx slider
+        0x57AE36,            // 35 Draw distance slider
+        0x57AE40,            // 36 Draw distance slider
+        0x57B064,            // 37 Mouse acc slider
+        0x57B06E,            // 38 Mouse acc slider
+        0x584A12,            // 39 DrawYouAreHereSprite
+        0x5740BC,            // 40 Message screen
+    };
+
+    int m_dwFrontendHeight[] = {
+        0x5795AD,            // 0 Menu text
+        0x579958,            // 1 Menu text
+        0x5799D5,            // 2 Menu text
+        0x57A345,            // 3 Menu text
+        0x57A2DB,            // 4 Menu text
+        0x57A25F,            // 5 Menu text
+        0x5796F6,            // 6 Menu text
+        0x576398,            // 7 Menu text
+        0x57497D,            // 8 Menu text
+        0x574FED,            // 9 Menu text
+        0x579643,            // 10 Menu text
+        0x579841,            // 11 Menu text
+        0x57A1AE,            // 12 Menu text
+        0x57A3C5,            // 13 Menu text
+        0x582DC1,            // 14 Legend
+        0x582ED3,            // 15 Legend
+        0x582FA3,            // 16 Legend
+        0x582FF5,            // 17 Legend
+        0x583079,            // 18 Legend
+        0x5830F6,            // 19 Legend
+        0x58317B,            // 20 Legend
+        0x576119,            // 21 Legend
+        0x575EB7,            // 22 Map zones
+        NULL,                // 23
+        NULL,                // 24
+        NULL,                // 25
+        NULL,                // 26
+        NULL,                // 27
+        NULL,                // 28
+        NULL,                // 29
+        NULL,                // 30
+        NULL,                // 31
+        NULL,                // 32
+        NULL,                // 33
+        NULL,                // 34
+        NULL,                // 35
+        NULL,                // 36
+        NULL,                // 37
+        0x5849FA,            // 38 DrawYouAreHereSprite
+        0x574096,            // 39 Message screen
+    };
+
+    for (int i = 0; i < sizeof(m_dwFrontendWidth) / sizeof(const void*); i++)
+    {
+        if (m_dwFrontendWidth[i] != NULL)
+            MemPut<const void*>(m_dwFrontendWidth[i] + 0x2, &fFrontendWidth[i]);
```

Comment:
The pointer type here should be float because the values ​​at these addresses are of type float. In addition, why do you pass the address of the value to memory, instead of the values ​​themselves?